kyuubi-server/src/main/scala/org/apache/kyuubi/engine/flink/FlinkProcessBuilder.scala

```diff
@@ -213,6 +254,32 @@ class FlinkProcessBuilder(
     }
   }
 
+  @volatile private var tokenTempDir: java.nio.file.Path = _
+  private def generateTokenFile(): Option[(String, String)] = {
+    // We disabled `hadoopfs` token service, which may cause yarn client to miss hdfs token.
+    // So we generate a hadoop token file to pass kyuubi engine tokens to submit process.
+    // TODO: Removed this after FLINK-35525, delegation tokens will be passed by `kyuubi` provider
+    conf.getOption(KYUUBI_ENGINE_CREDENTIALS_KEY).map { encodedCredentials =>
+      val credentials = KyuubiHadoopUtils.decodeCredentials(encodedCredentials)
+      tokenTempDir = Utils.createTempDir()
```

Comment:
In Spark, `spark-submit` takes responsibility for cleaning the local staging dir on exiting. Does `flink` have a similar mechanism?